Tests/Exporter/ExporterTest.php

```diff
@@ -12,21 +12,61 @@
 namespace Sonata\CoreBundle\Tests\Exporter;
 
 use Exporter\Source\ArraySourceIterator;
+use Exporter\Writer\CsvWriter;
+use Exporter\Writer\JsonWriter;
+use Exporter\Writer\XlsWriter;
+use Exporter\Writer\XmlWriter;
 use Sonata\CoreBundle\Exporter\Exporter;
 
 class ExporterTest extends \PHPUnit_Framework_TestCase
 {
-    /**
-     * @expectedException RuntimeException
-     */
     public function testFilter()
     {
+        $this->setExpectedException('RuntimeException', 'Invalid "foo" format');
         $source = $this->getMock('Exporter\Source\SourceIteratorInterface');
+        $writer = $this->getMock('Exporter\Writer\WriterInterface');
 
-        $exporter = new Exporter();
+        $exporter = new Exporter(array('mywriter' => $writer));
         $exporter->getResponse('foo', 'foo', $source);
     }
 
+    public function testConstructorRejectsNonWriters()
+    {
+        $this->setExpectedException('InvalidArgumentException', 'Writer for format "made-up-format"');
+        new Exporter(array('made-up-format' => 'Not even an object'));
+    }
+
+    /**
+     * NEXT MAJOR: remove this test.
```

Comment:
`NEXT_MAJOR`

https://github.com/sonata-project/SonataCoreBundle/blob/3.x/CONTRIBUTING.md#the-base-branch